pkg/i2gw/ingress2gateway.go

```diff
@@ -198,3 +199,74 @@ func mergeGateways(gatewaResources []GatewayResources) (map[types.NamespacedName
 
 	return newGateways, errs
 }
+
+// MergeIRs accepts multiple IRs and creatse a unique IR struct built
+// as follows:
+//   - GatewayClasses, Routes, and ReferenceGrants are grouped into the same maps
+//   - Gateways may have the same NamespaceName even if they come from different
+//     ingresses, as they have a their GatewayClass' name as name. For this reason,
+//     if there are mutiple gateways named the same, their listeners are merged into
+//     a unique Gateway.
+//
+// This behavior is likely to change after https://github.com/kubernetes-sigs/gateway-api/pull/1863 takes place.
+func MergeIRs(irs ...IR) (IR, field.ErrorList) {
+	mergedIRs := IR{
+		Gateways:        make(map[types.NamespacedName]GatewayContext),
+		GatewayClasses:  make(map[types.NamespacedName]gatewayv1.GatewayClass),
+		HTTPRoutes:      make(map[types.NamespacedName]HTTPRouteContext),
+		Services:        make(map[types.NamespacedName]*ServiceIR),
+		TLSRoutes:       make(map[types.NamespacedName]gatewayv1alpha2.TLSRoute),
+		TCPRoutes:       make(map[types.NamespacedName]gatewayv1alpha2.TCPRoute),
+		UDPRoutes:       make(map[types.NamespacedName]gatewayv1alpha2.UDPRoute),
+		ReferenceGrants: make(map[types.NamespacedName]gatewayv1beta1.ReferenceGrant),
+	}
+	var errs field.ErrorList
+	mergedIRs.Gateways, errs = mergeGatewayContexts(irs)
+	if len(errs) > 0 {
+		return IR{}, errs
+	}
+	for _, gr := range irs {
+		maps.Copy(mergedIRs.GatewayClasses, gr.GatewayClasses)
+		maps.Copy(mergedIRs.HTTPRoutes, gr.HTTPRoutes)
```

Comment:
I re-added the comment on merging httpIRs and ServiceIRs here https://github.com/kubernetes-sigs/ingress2gateway/pull/185#discussion_r1745843418

Please either add logic to merge them or add a clear TODO plus an i2gw issue to implement that

Reply:
Done. Added an issue to track this.